src/Microsoft.DotNet.Build.Tasks/GetTargetMachineInfo.cs

```diff
@@ -29,7 +48,7 @@ public override bool Execute()
             else if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("NETBSD")))
                 TargetOS = "NetBSD";
 
-            return true;
+            return TargetArch != null && TargetOS != null;
```

Comment:
Probably worth an error message here as well saying which one is unknown or not supported.